src/apify/_actor.py

```diff
@@ -184,18 +187,21 @@ async def init(self) -> None:
         if self._is_initialized:
             raise RuntimeError('The Actor was already initialized!')
 
-        if self._configuration.token:
-            service_container.set_cloud_storage_client(ApifyStorageClient(configuration=self._configuration))
+        self._is_exiting = False
+        self._was_final_persist_state_emitted = False
 
-        if self._configuration.is_at_home:
-            service_container.set_default_storage_client_type('cloud')
+        # Register services in the service locator.
+        if self.is_at_home():
+            service_locator.set_storage_client(self._cloud_storage_client)
         else:
-            service_container.set_default_storage_client_type('local')
+            service_locator.set_storage_client(self._local_storage_client)
```

Comment:
This may unnecessarily override a user-defined storage client when running locally - they could have some specific settings in place, or a custom implementation. I believe we could just remove this branch completely.

Reply:
yep, let's remove setting the local storage client in Actor

(for local event manager this is unfortunately not possible, because of the missing config-init)